projects/ngx-speculoos/src/lib/route.spec.ts

```diff
@@ -119,19 +135,220 @@ describe('routes', () => {
+      expect(route.snapshot.parent).toBe(parent.snapshot);
+      expect(route.snapshot.children).toEqual(children.map(c => c.snapshot));
+      expect(route.snapshot.firstChild).toBe(firstChild.snapshot);
+      expect(route.snapshot.pathFromRoot).toEqual([parent.snapshot, route.snapshot]);
+      expect(route.snapshot.root).toBe(parent.snapshot);
+
+      let params: Params;
+      route.params.subscribe(p => (params = p));
+      expect(params).toBe(providedParams);
+
+      let paramMap: ParamMap;
+      route.paramMap.subscribe(p => (paramMap = p));
+      expect(paramMap.get('foo')).toBe('bar');
+
+      let queryParams: Params;
+      route.queryParams.subscribe(p => (queryParams = p));
+      expect(queryParams).toBe(providedQueryParams);
+
+      let queryParamMap: ParamMap;
+      route.queryParamMap.subscribe(p => (queryParamMap = p));
+      expect(queryParamMap.get('baz')).toBe('bing');
+
+      let fragment: string;
+      route.fragment.subscribe(p => (fragment = p));
+      expect(fragment).toBe(providedFragment);
+
+      let data: Data;
+      route.data.subscribe(p => (data = p));
+      expect(data).toBe(providedData);
+
+      let url: Array<UrlSegment>;
+      route.url.subscribe(p => (url = p));
+      expect(url).toBe(providedUrl);
+
+      expect(route.parent).toBe(parent);
+      expect(route.children).toBe(children);
+      expect(route.firstChild).toBe(firstChild);
+      expect(route.pathFromRoot).toEqual([parent, route]);
+      expect(route.root).toBe(parent);
+    });
+
+    it('should set a param', () => {
+      const route = stubRoute({ params: { a: 'a1', b: 'b1' } });
+
+      const expectedParams = { a: 'a2', b: 'b1' };
+
+      let queryParamEmissionCount = 0;
+      let snapshotAtEmissionTime: ActivatedRouteSnapshot;
+      route.queryParams.subscribe(() => queryParamEmissionCount++);
+      let params: Params;
+      route.params.subscribe(p => {
+        params = p;
+        snapshotAtEmissionTime = route.snapshot;
+      });
+
+      route.setParam('a', 'a2');
+
+      expect(route.snapshot.params).toEqual(expectedParams);
+      expect(snapshotAtEmissionTime.params).toEqual(expectedParams);
+      expect(params).toEqual(expectedParams);
+      expect(queryParamEmissionCount).toBe(1);
+    });
+
+    it('should set a query param', () => {
+      const route = stubRoute({ queryParams: { a: 'a1', b: 'b1' } });
+
+      const expectedQueryParams = { a: 'a2', b: 'b1' };
+      let queryParams: Params;
+      route.queryParams.subscribe(p => (queryParams = p));
+
+      route.setQueryParam('a', 'a2');
+
+      expect(route.snapshot.queryParams).toEqual(expectedQueryParams);
+      expect(queryParams).toEqual(expectedQueryParams);
+    });
+
+    it('should set a datum', () => {
+      const route = stubRoute({ data: { a: 'a1', b: 'b1' } });
+
+      const expectedData = { a: 'a2', b: 'b1' };
+      let data: Data;
+      route.data.subscribe(p => (data = p));
+
+      route.setDatum('a', 'a2');
```

Comment:
setData is already used for setting all the data at once. And I'd prefer a different method name rather than an overload, to be consistent, so I'll use setDataItem()